site/app/templates/AutogradingStatus.twig

```diff
@@ -0,0 +1,192 @@
+                <th class="col-md-2" colspan="{{ progress["capability_queue_counts"]|keys|length }}" style="text-align: left;">
+                    CAPABILITIES (queue)
+                </th>
+            </tr>
+            <tr>
+                <th class="col-md-2" rowspan="2">
+                    grading
+                </th>
+                <th class="col-md-2 right-boarder" rowspan="2">
+                    in queue
+                </th>
+                <th class="col-md-2" rowspan="2">
+                    grading
+                </th>
+                <th class="col-md-2 right-boarder" rowspan="2">
+                    in queue
+                </th>
+                {% for machines in progress["machine_grading_counts"]|keys %}
+                {% if machines == progress["machine_grading_counts"]|keys|last %}
+                <th class="col-md-2 right-boarder">
+                    {{ machines|upper }}
+                </th>
+                {% else %}
+                <th class="col-md-2">
+                    {{ machines|upper }}
+                </th>
+                {% endif %}
+                {% endfor %}
+                {% for capability in progress["capability_queue_counts"]|keys %}
+                <th class="col-md-2" rowspan="2">
+                    {{ capability|upper }}
+                </th>
+                {% endfor %}
+            </tr>
+            <tr>
+                {# Should display number of shippers instead for each machine #}
+                {% for machine in progress["machine_grading_counts"]|keys %}
+                    {% if machine == progress["machine_grading_counts"]|keys|last %}
+                    <th class="right-boarder">
+                        {{ progress["num_autograding_workers"][machine] }}
+                    </th>
+                    {% else %}
+                    <th>
+                        {{ progress["num_autograding_workers"][machine] }}
+                    </th>
+                    {% endif %}
+                {% endfor %}
+                <th colspan="{{ progress["capability_queue_counts"]|keys|length }}">
+                </th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td class="right-boarder">
+                    {{ progress["time"] }}
+                </td>
+                <td>
+                    {{ progress["queue_counts"]["interactive_ongoing"] }}
+                </td>
+                <td class="right-boarder">
+                    {{ progress["queue_counts"]["interactive"] }}
+                </td>
+                <td>
+                    {{ progress["queue_counts"]["regrade_ongoing"] }}
+                </td>
+                <td class="right-boarder">
+                    {{ progress["queue_counts"]["regrade"] }}
+                </td>
+                {% for machine, count in progress["machine_grading_counts"] %}
+                {% if machine == progress["machine_grading_counts"]|keys|last %}
```

Comment:
It might be cleaner if this section was indented a little bit.

Reply:
Good point, I never know how twig should look like styling wise. I have seen both indenting and not indenting done but I was never sure.